storage/zynq7000-sdcard/sdcard.c

```diff
@@ -0,0 +1,1268 @@
+		return -ENOMEM;
+	}
+
+	host->dmaBufferPhys = sdhost_mphys(host->dmaBuffer, &psize);
+
+	if (SDCARD_MAX_TRANSFER != psize) {
+		munmap(host->dmaBuffer, SDCARD_MAX_TRANSFER);
+		host->dmaBuffer = NULL;
+		return -1;
+	}
+
+	if (host->dmaBufferPhys & 3) {
+		munmap(host->dmaBuffer, SDCARD_MAX_TRANSFER);
+		host->dmaBuffer = NULL;
+		return -1;
+	}
+
+	return 0;
+}
+
+
+static int sdhost_isr(unsigned int n, void *arg)
+{
+	/* SD Host controller interrupts are triggered by level, not by edge.
+	 * This means that ISR has to remove the reason for interrupt (zero out bit in STATUS register)
+	 * or disable the interrupt (zero out bit in SIGNAL_ENABLE register) - otherwise we get stuck
+	 * in infinite interrupt loop.
+	 */
+	sdcard_hostData_t *host = (sdcard_hostData_t *)arg;
+	uint32_t activeInterrupts = *(host->base + SDHOST_REG_INTR_STATUS);
+	*(host->base + SDHOST_REG_INTR_SIGNAL_ENABLE) &= ~activeInterrupts;
+	return 0;
+}
+
+
+static int _sdio_cmdExecutionWait(sdcard_hostData_t *host, uint32_t flags, time_t wait_us)
+{
+	uint32_t val, ret = -ETIME;
+	uint32_t flagsWithErrors = flags | SDHOST_ERROR_REASONS;
+
+	int doResetCmd = 0, doResetDat = 0;
+
+	*(host->base + SDHOST_REG_INTR_SIGNAL_ENABLE) |= flagsWithErrors | SDHOST_INTR_CARD_OUT;
+	/* If there is a pending interrupt from flagsWithErrors set, it will fire now */
+	int waitret = condWait(host->eventCond, host->eventLock, wait_us);
+	if (waitret == -ETIME) {
+		/* It's a bit odd, but card ejection doesn't cause any errors - only timeouts
+		 * We may get a SDHOST_INTR_CARD_OUT, but only if physical card detect is available
+		 */
+		host->card.commandTimeouts++;
+	}
+
+	val = *(host->base + SDHOST_REG_INTR_STATUS);
+	if (val & SDHOST_ERROR_REASONS) {
+		doResetCmd = (val & SDHOST_INTR_CMD_ERRORS) ? 1 : 0;
+		doResetDat = (val & SDHOST_INTR_DAT_ERRORS) ? 1 : 0;
+		*(host->base + SDHOST_REG_INTR_STATUS) = SDHOST_ERROR_REASONS;
+		ret = -EIO;
+	}
+	else if ((val & flags) == flags) {
+		*(host->base + SDHOST_REG_INTR_STATUS) = flags;
+		ret = EOK;
+	}
+
+	if (doResetCmd || ret == -ETIME) {
+		sdhost_reset(host, CLOCK_CONTROL_RESET_CMD);
+	}
+
+	if (doResetDat || ret == -ETIME) {
+		sdhost_reset(host, CLOCK_CONTROL_RESET_DAT);
+	}
+
+	if (val & SDHOST_INTR_CARD_OUT) {
+		condSignal(presenceEvents.cond);
+	}
+
+	*(host->base + SDHOST_REG_INTR_SIGNAL_ENABLE) &= ~flagsWithErrors;
+	return ret;
+}
+
+
+/* Internal command sending function.
+ * @param cmd Command index (one of SDIO_CMD_* or SDIO_ACMD_*)
+ * @param arg Command argument
+ * @param res Response output. If isLongResponse == 0, a single uint32_t is expected. If isLongResponse == 1, an array of 4 uint32_t is expected.
+ * @param blockCount How many blocks of data are transferred in this command
+ * @param isLongResponse Expected response length (see parameter `res`)
+ */
+static int _sdio_cmdSend(
+	sdcard_hostData_t *host,
+	uint8_t cmd,
+	uint32_t arg,
+	uint32_t *res,
+	uint16_t blockCount,
+	int isLongResponse)
+{
```

Comment:
We don't use this style
```suggestion
static int _sdio_cmdSend( sdcard_hostData_t *host, uint8_t cmd, uint32_t arg, uint32_t *res, uint16_t blockCount, int isLongResponse)
{
```